File: packages/react-native-fast-io/ios/HybridStreamFactory.swift
//
//  HybridStreamFactory.swift
//  FastIO
//
//  Created by Mike Grabowski on 12/11/2024.
//

import Foundation
import FastIOPrivate

class HybridStreamFactory : HybridStreamFactorySpec {
  static let BUFFER_SIZE = FASTIO_BUFFER_SIZE
  
  var bufferSize: Double {
    get {
      Double(HybridStreamFactory.BUFFER_SIZE)
    }
  }
  
  func createInputStream(path: String) -> any HybridInputStreamSpec {
    guard let stream = InputStream(fileAtPath: path) else {
      fatalError("Failed to create stream from \(path)")
    }
    return HybridInputStream(stream: stream)
  }
  
  var hybridContext = margelo.nitro.HybridContext()
  var memorySize: Int {
    return getSizeOf(self)
  }
}

File: packages/react-native-fast-io/android/src/main/java/com/margelo/nitro/fastio/HybridStreamFactory.kt
package com.margelo.nitro.fastio

import com.margelo.nitro.core.Promise

class HybridStreamFactory : HybridStreamFactorySpec() {
    override val bufferSize: Double
        get() = throw NotImplementedError("HybridStreamFactory.bufferSize not implemented")

    override fun createInputStream(path: String): HybridInputStreamSpec {
        throw NotImplementedError("HybridStreamFactory.createInputStream() not implemented")
    }

    override val memorySize: Long
        get() = 0L
} 
